sdk/core/azure-core/src/datetime.cpp

```diff
@@ -127,19 +133,20 @@ constexpr char const monthNames[] = "Jan\0Feb\0Mar\0Apr\0May\0Jun\0Jul\0Aug\0Sep
 
 } // namespace
 
-std::string DateTime::ToString(DateFormat format, TimeFractionFormat fractionFormat) const
+std::string DateTime::GetString(DateFormat format, TimeFractionFormat fractionFormat) const
 {
-  if (m_interval > 2650467743999999999LL)
+  static constexpr auto const EndOfYear9999 = 2650467743999999999LL;
+  if (m_since1601.count() > EndOfYear9999)
   {
-    throw DateTimeException("The requested year exceeds the year 9999.");
+    throw std::length_error("The requested year exceeds the year 9999.");
```

Comment:
format error instead?

Reply:
+1 -- length_error is "you tried to form a container whose length is unrepresentable" e.g. `std::string("hello").append(SIZE_MAX, 'a')`

Reply:
format_error is C++20.

Reply:
Missed that. `runtime_error` then? The main point is, pick a reasonable exception type here, that fits, other than `length_error`.

Reply:
I'd probably make your own type for this (IIRC at one time there was a plan for a shared az_core exception hierarchy? I don't think this integer overflow case matches any of the existing ones well)

Reply:
format_error is C++20. we have 3 candidates - length_error, out_of_range, and invalid_argument. I did not want to throw invalid_argument when it is not based on the method input parameter (assuming `this` is not one of them).
I see if you don't like `length_error`, you probably won't like `out_of_range` either, so let me to throw only `invalid_argument`.

Reply:
@BillyONeal, yesterday we haven't decided finally, but leaning heavily towards not having the AzureSDK exception hierarchy, and even reusing std::exception where suitable. It would've been great if you participated in that discussion. But so far, yesterday we decided on throwing plain std:: exceptions in this case.